addon/adapters/pouch.js

```diff
@@ -0,0 +1,161 @@
+import Ember from 'ember';
+import DS from 'ember-data';
+import PouchDB from 'pouchdb';
+
+import {
+  extractFindOne,
+  extractDeleteRecord,
+  updateSchemaForType
+} from 'ember-pouch/utils';
+
+var bind = Ember.run.bind;
+var next = Ember.run.next;
+
+export default DS.Adapter.extend({
+  defaultSerializer: '-pouch',
+  coalesceFindRequests: true,
+  schema: null,
+  url: null,
+
+  setup: function(url) {
+    url = url || this.url;
+
+    if (url) { this.db = new PouchDB(url); }
+  }.on('init'),
```

Comment:
@stefanpenner has anyone considered detecting if super was called or not, and then intelligently calling it for the user when necessary? I just noticed I had some js library code that does this exact thing